src/platform/nrfconnect/NrfconnectUtils.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "NrfconnectUtils.h"
+
+namespace chip {
+namespace DeviceLayer {
+
+CHIP_ERROR HexStringToByteSpan(const char * input, size_t size, MutableByteSpan & output)
+{
+    ReturnErrorCodeIf(!input, CHIP_ERROR_INVALID_ARGUMENT);
+    ReturnErrorCodeIf((size / 2) > output.size(), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    auto decode = [](char value) -> uint8_t {
+        if (value >= '0' && value <= '9')
+            return static_cast<uint8_t>(value - '0');
+        if (value >= 'A' && value <= 'F')
+            return static_cast<uint8_t>(value - 'A' + 10);
+        if (value >= 'a' && value <= 'f')
+            return static_cast<uint8_t>(value - 'a' + 10);
+        return 0;
+    };
+
+    auto createInt = [decode](const char * value) -> uint8_t {
+        return static_cast<uint8_t>(decode(value[0]) * 16 + decode(value[1]));
```

Comment:
This will read out of bounds at the end if `size` is odd, right?

More to the point, this whole function just does the same thing as `chip::Encoding::HexToBytes(input, size, output.data(), output.size())`, no?

Reply:
Oh, I didn't realize that there is a function to that conversion within the Matter repo. I will use `HexToBytes` to not duplicate the code. Thanks for spotting that. :)